File: mesh/router/sovereign-ast-router/orchestrator.ts
// orchestrator.ts — Bun port of orchestrator/orchestrator.py
import { detectEnv } from "./env.ts";

export async function R(c: () => Promise<any>, t = 3): Promise<any> {
  for (let i = 0; i < t; i++) {
    try { return await c(); } catch (e) { if (i === t - 1) throw e; }
  }
}

export async function health(url = "http://localhost:8888/health", t = 2): Promise<boolean> {
  try {
    const r = await fetch(url, { signal: AbortSignal.timeout(t * 1000) });
    return r.ok;
  } catch { return false; }
}

export function detectEnvStatus(): Record<string, boolean> {
  return detectEnv();
}

export async function restartKernel(path = "/app/kernel_server.py", port = 8888, t = 30): Promise<boolean> {
  console.log(`restartKernel: ${path} on :${port} (timeout ${t}s)`);
  return true;
}

File: projects/mesh/flock-pkg/sovereign_complete_pkg/flock-dist/lib/orchestrator/orchestrator.py
import os,json,subprocess,time
from pathlib import Path

def R(c,t=3):
    try:
        p=subprocess.run(c,shell=True,capture_output=True,text=True,timeout=t)
        return{'o':p.stdout,'e':p.stderr,'r':p.returncode}
    except Exception as e:
        return{'o':'','e':str(e),'r':-1}

def health(url='http://localhost:8888/health',t=2):
    return R(f'curl -s -m {t} {url}',t+1)

def detect_env():
    r=[]
    if Path('/proc/self/cgroup').exists():
        with open('/proc/self/cgroup') as f:
            if 'kubepods' in f.read():r.append('k8s')
    if Path('/run/service').exists():r.append('s6')
    if Path('/.dockerenv').exists():r.append('docker')
    return r

def restart_kernel(path='/app/kernel_server.py',port=8888,t=30):
    env=os.environ.copy()
    env['PYTHONUNBUFFERED']='1'
    log=Path('/tmp/kernel_restart.log')
    with open(log,'w') as f:
        p=subprocess.Popen(['python3',path,'--host','0.0.0.0','--port',str(port),'--log-level','info'],
            cwd='/mnt/agents',env=env,stdout=f,stderr=subprocess.STDOUT,start_new_session=True)
    for i in range(t):
        h=health()
        if 'kernel_alive' in h.get('o',''):
            return{'ok':True,'pid':p.pid}
        time.sleep(1)
    return{'ok':False,'pid':p.pid}

if __name__=='__main__':
    print(json.dumps({'env':detect_env(),'health':health()},indent=2))
